src/core/src/service_interfaces/StatusHandler.py

```diff
@@ -624,22 +613,36 @@ def load_status_file_components(self, initial_load=False):
                     self.__assessment_total_error_count = self.__get_total_error_count_from_prev_status(errors['message'])
             if name == Constants.PATCH_METADATA_FOR_HEALTHSTORE:     # if it exists, it must be to spec, or an exception will get thrown
                 if self.execution_config.exec_auto_assess_only:
-                    self.__metadata_for_healthstore_substatus_json = status_file_data['status']['substatus'][i]
+                    self.__metadata_for_healthstore_substatus_json = complete_status_file_data['status']['substatus'][i]
                 else:
-                    message = status_file_data['status']['substatus'][i]['formattedMessage']['message']
+                    message = complete_status_file_data['status']['substatus'][i]['formattedMessage']['message']
                     self.__metadata_for_healthstore_summary_json = json.loads(message)
             if name == Constants.CONFIGURE_PATCHING_SUMMARY:     # if it exists, it must be to spec, or an exception will get thrown
                 if self.execution_config.exec_auto_assess_only:
-                    self.__configure_patching_substatus_json = status_file_data['status']['substatus'][i]
+                    self.__configure_patching_substatus_json = complete_status_file_data['status']['substatus'][i]
                 else:
-                    message = status_file_data['status']['substatus'][i]['formattedMessage']['message']
+                    message = complete_status_file_data['status']['substatus'][i]['formattedMessage']['message']
                     self.__configure_patching_summary_json = json.loads(message)
                     errors = self.__configure_patching_summary_json['errors']
                     if errors is not None and errors['details'] is not None:
                         self.__configure_patching_errors = errors['details']
                         self.__configure_patching_top_level_error_count = self.__get_total_error_count_from_prev_status(errors['message'])
 
-    def __write_status_file(self):
+    def __load_complete_status_file_data(self, file_path):
+        # Read the status file - raise exception on persistent failure
+        for i in range(0, Constants.MAX_FILE_OPERATION_RETRY_COUNT):
+            try:
+                with self.env_layer.file_system.open(file_path, 'r') as file_handle:
+                    complete_status_file_data = json.load(file_handle)[0]    # structure is array of 1
+            except Exception as error:
+                if i < Constants.MAX_FILE_OPERATION_RETRY_COUNT - 1:
+                    time.sleep(i + 1)
+                else:
+                    self.composite_logger.log_error("Unable to read status file (retries exhausted). Error: {0}.".format(repr(error)))
+                    raise
+        return complete_status_file_data
+
+    def __write_complete_status_file(self):
```

Comment:
nit: the name __write_complete_status_file() suggests it only writes <>.complete.status file whereas this function writes both <>.complete.status and <>.status file

suggestion: rename this to __write_status_files()

Reply:
@feng-j678, I'm approving this PR. It's okay if this PR goes in without this change, but please make sure to include this comment in your next PR